### PR TITLE
Feat: Added delete button for items that connects to database

### LINKED ISSUE #11: 11. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future.
It’s possible our users may make an error when entering an item or may decide they won’t be buying a certain item again in the future. Let’s help them keep their lists tidy!

AC:

* User is able to delete an item from the shopping list
* Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
* Deletion should cause the associated record(s) in the database to be deleted

### PR DESCRIPTION
**Pull request by audracodes**
## Description

This PR adds a delete button to each item in a user's shopping list. Upon clicking the delete button the user is prompted to ensure that they'd like to delete the item, if they click `ok` then the item is removed from their list and the database. 

## Related Issue

Closes #11 

## Acceptance Criteria

- [x] User is able to delete an item from the shopping list
- [x] Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
-[x] Deletion should cause the associated record(s) in the database to be deleted

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
| ✓| :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

**ViewList**

Added a `button` to the `li` of the rendered user list, when the button is clicked `handleDelete` runs. `handleDelete` uses `confirm` to show a prompt to the user asking if they'd like to delete their item. If the user clicks `ok` then the item is removed from the database. 

The checkbox input from the list was also moved into the `li` to make the UI more straight forward.

### Before

Screen Shot 2020-11-10 at 3 36 25 PM

### After

Screen Shot 2020-11-10 at 3 35 46 PM

Screen Shot 2020-11-10 at 3 35 53 PM

## Testing Steps / QA Criteria

Make sure you are viewing a list that contains items. In the `ViewList` component, each list item should have a delete button beside the name of the item. Upon clicking delete a prompt should appear asking for confirmation of deletion. After clicking 'ok' the item should be removed from the list and from the database.

### CODE REVIEW

**Comment by rachel-fischoff** [comment]
@snsebro @rbompiani ready for review!

File: src/components/ViewList.js [hunk 1]
```diff
@@ -70,6 +70,19 @@ const ViewList = () => {
     setFilterValue(e.target.value);
   };
 
+  const handleDelete = (e) => {
+    window.confirm('Would you like to delete your item?');
```
Comment (rbompiani) [inline src/components/ViewList.js]: Clicking "cancel" in the confirm still deletes the item from the database.  We'll need something here to conditionally delete based on the confirm response.

File: src/components/ViewList.js [hunk 2]
```diff
@@ -93,20 +106,26 @@ const ViewList = () => {
...
-                type="checkbox"
-                name={element.itemName}
-                id={element.id}
-                value={element.itemName}
-                className="purchased"
-                onChange={handleCheck}
-                checked={element.isPurchased}
-              ></input>
-              <li> {element.itemName} </li>
+              <li>
```
Comment (rbompiani) [inline src/components/ViewList.js:116]: I like this nesting a lot more than what we had previously. Shayne and I had a few edits here as well (styling, mostly). When merging we should keep this nesting but move class styling from the other issue.

**Review by rbompiani — CHANGES_REQUESTED** [review]
Looks good!  Just needs a catch to divert from deleting the item from the database if the user cancels via the confirm window.

**Review by rbompiani — APPROVED** [review]

**Review by indiragp — APPROVED** [review]
Great Job!

**Pull request merged**